service/bblfsh/bblfsh.go

```diff
@@ -42,22 +44,95 @@ func (s *Service) GetChanges(ctx context.Context, req *lookout.ChangesRequest) (
 	}
 
 	return &ChangeScanner{
-		underlying:    changes,
-		ctx:           ctx,
-		client:        s.client,
-		purgeContents: !wantContents,
+		underlying: changes,
+		BaseScanner: BaseScanner{
+			ctx:           ctx,
+			client:        s.client,
+			purgeContents: !wantContents,
+		},
 	}, nil
 }
 
-type ChangeScanner struct {
-	underlying    lookout.ChangeScanner
+func (s *Service) GetFiles(ctx context.Context, req *lookout.FilesRequest) (
+	lookout.FileScanner, error) {
+	wantContents := req.WantContents
+	if req.WantUAST {
+		req.WantContents = true
+	}
+
+	files, err := s.files.GetFiles(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+
+	if !req.WantUAST {
+		return files, nil
+	}
+
+	return &FileScanner{
+		underlying: files,
+		BaseScanner: BaseScanner{
+			ctx:           ctx,
+			client:        s.client,
+			purgeContents: !wantContents,
+		},
+	}, nil
+}
+
+type BaseScanner struct {
 	ctx           context.Context
 	client        protocol.ProtocolServiceClient
 	purgeContents bool
-	val           *lookout.Change
 	err           error
 }
 
+func (s *BaseScanner) processFile(f *lookout.File) error {
+	if f == nil {
+		return nil
+	}
+
+	var err error
+	f.UAST, err = s.parseFile(f)
+	if err != nil {
+		return err
+	}
+
+	if s.purgeContents {
+		f.Content = nil
+	}
+
+	return nil
+}
+
+func (s *BaseScanner) parseFile(f *lookout.File) (
+	*uast.Node, error) {
```

Comment:
Genuinely interested, is there a convention or styleguide for formatting it like, this instead of

```
func (s *BaseScanner) parseFile(f *lookout.File) (*uast.Node, error) {
}
```
?

Reply:
This style is used only when the line is too long. Go doesn't have specific styleguide for it:
https://github.com/golang/go/wiki/CodeReviewComments#line-length

but in this particular case, the function isn't long. So it should be one-liner. 👍